odoo/addons/base/models/assetsbundle.py

```diff
@@ -198,12 +223,12 @@ def checksum(self):
         return hashlib.sha512(check.encode('utf-8')).hexdigest()[:64]
 
     def _get_asset_template_url(self):
-        return "/web/content/{id}-{unique}/{extra}{name}{sep}{type}"
+        return "/web/content/{id}{unique}/{extra}{name}{sep}{type}"
 
     def _get_asset_url_values(self, id, unique, extra, name, sep, type):  # extra can contain direction or/and website
         return {
             'id': id,
-            'unique': unique,
+            'unique': f'-{unique}' if unique else '',
```

Comment:
Perhaps I missed something, but this seems to break the *debug=assets* mode as soon as `website` is installed?

In https://github.com/odoo/odoo/blob/master/addons/website/models/ir_qweb.py#L19-L22 it adds the website ID in the asset URL, using a trick with the `extra` value.
So instead of this: `/web/content/1092/web.assets_common_minimal_js.js` the result ends up looking like this: `/web/content/1092/1/web.assets_common_minimal_js.js`

But this means [the `/web/content` route matching](https://github.com/odoo/odoo/blob/master/addons/web/controllers/main.py#L1432-L1442) can now only match `/web/content/<string:model>/<int:id>/<string:field>` instead of `/web/content/<int:id>/<string:filename>`:
- `model`: `1092`
- `id`: `1`
- `field`: `web.assets_common_minimal_js.js`

This should never match anything -> 404.

We don't want to cause more side-effects by changing those route patterns, so I see a few options:
- keep the `{id}-{unique}` using a hardcoded unique, e.g. `debug`, instead of dropping it, to preserve the match priorities
- change the trick in `website` to avoid the ambiguity
- use a dedicated route for assets, e.g. `/web/assets/...`, similarly to what we do with `/web/image/*`, so that we can make the parameters more specific, yet delegate the actual work to `content_common()`?

The third option is perhaps the most elegant one, because assets are clearly much more than static attachments, and may need to get more special treatment in the future.

Reply:
hmm, I wasn't aware of that. This means that we probably miss a test that loads website assets in debug mode.  

I like your first solution, with `debug` as a hardcoded unique value. Very simple. But solution 3 sounds good as well.